### ISSUE TITLE
openURL, openEditDialog, openCancelDialog not defined

### ISSUE THREAD

**Issue by araz-hashemi** [llazzaro/django-scheduler#236]
When clicking on the edit/cancel icon of an event, it pulls up the preview slide-down (not the edit/cancel form) and the console logs an Uncaught ReferenceError : openURL/openCancelDialog not defined 

e.g. see generated source

```
<span class="actions">
  <span class="edit">
        <a href="#" onclick="openEditDialog(this, '/schedule/occurrence/edit/1/2016/8/16/11/0/0/?next=/schedule/calendar/week/ksat-stream-1/%3Fyear%3D2016%26month%3D8%26day%3D17', '/schedule/event/edit/ksat-stream-1/1/?next=/schedule/calendar/week/ksat-stream-1/%3Fyear%3D2016%26month%3D8%26day%3D17', event);">
          <span class="glyphicon glyphicon-pencil"></span>
        </a>
  </span>

<span class="delete">
        <a href="#" onclick="openCancelDialog(this, '/schedule/occurrence/cancel/1/2016/8/16/11/0/0/?next=/schedule/calendar/week/ksat-stream-1/%3Fyear%3D2016%26month%3D8%26day%3D17', '/schedule/event/delete/1/?next=/schedule/calendar/week/ksat-stream-1/%3Fyear%3D2016%26month%3D8%26day%3D17', event);">
          <span class="glyphicon glyphicon-remove"></span>
        </a>
</span>

```

Where are these functions defined? I only see them referenced in [_event_options.html|https://github.com/llazzaro/django-scheduler/blob/develop/schedule/templates/schedule/_event_options.html]

I'm pretty new to development, so I don't know if I'm missing something obvious or if there is an actual bug.

**Comment by llazzaro** [comment]
@araz-hashemi did you find a solution to this issue?

**Comment by araz-hashemi** [comment]
@llazzaro : Well, not really, but I assumed it might have been simple ignorance on my part. 
Are those functions part of some javascript dependency? 
I manually added the jQuery and Bootstrap static files, so perhaps I missed some components? Do the button actions work for you?

**Comment by llazzaro** [comment]
I will try to review this. You have to add jQuery and bootstrap. If I force a jQuery or bootstrap I can break something if you are using jQuery or Bootstrap.
I think that in the readme explains how to install js dependencies with django-bower.

**Comment by chrillux** [comment]
I experienced the same problems. From what I've seen, it looks like these functions were removed in commit:

https://github.com/llazzaro/django-scheduler/commit/e181737988fd750d28f68b6180bb87d1a4e3d0a4#diff-ea616e02916d349b42d0fb830adaf77e

If I put the removed functions in schedule/templates/schedule/_event_options.html, I could edit and delete one time events, i.e not a recurring event (! occurrence.event.rule):

```
diff --git a/schedule/templates/schedule/_event_options.html b/schedule/templates/schedule/_event_options.html
index ae2adc0..2d00aab 100644
--- a/schedule/templates/schedule/_event_options.html
+++ b/schedule/templates/schedule/_event_options.html
@@ -1,5 +1,35 @@
 {% load staticfiles i18n %}
 
+<script>
+function openCancelDialog(node, cancel_url, delete_url, event){
+  event.stopPropagation();
+  var btns = {"This":function(){window.location=cancel_url;}, "All":function(){window.location=delete_url}, "Do nothing":function(){$(this).dialog("de
+  dia = $("#delete_dialog").dialog({'buttons':btns, 'modal':true});
+  dia.dialog('open');
+  return false;
+}
+
+function openEditDialog(node, occurrence_url, event_url, event){
+  event.stopPropagation();
+  var btns = {"This":function(){window.location=occurrence_url;}, "All":function(){window.location=event_url}, "Do nothing":function(){$(this).dialog(
+  dia = $("#edit_dialog").dialog({'buttons':btns, 'modal':true});
+  dia.dialog('open');
+  return false;
+}
+
+function openDetail(node){
+  var btns = { "Close":function(){$(this).dialog("destroy");}};
+  dia = $($(node).attr("href")).dialog({'buttons':btns, 'modal':true, 'title':'Details'});
+  dia.dialog('open');
+  return false;
+}
+
+function openURL(url, event){
+    event.stopPropagation();
+    window.location=url;
+}
+</script>
+
```

Now when I clicked on edit or delete for a recurring event, nothing happened but in the web console I got this error:

`Uncaught TypeError: $(...).dialog is not a function`

Looks like JQuery UI is missing.

I downloaded JQuery UI from https://jqueryui.com/resources/download/jquery-ui-1.12.1.zip and put the jquery-ui.js in

`myapp/components/bower_components/jquery/dist`

Edit:

```
diff --git a/schedule/templates/base.html b/schedule/templates/base.html
index c549f31..60b3629 100644
--- a/schedule/templates/base.html
+++ b/schedule/templates/base.html
@@ -6,6 +6,7 @@
         <title>{% if site_name %}{{ site_name }} : {% endif %}{% block head_title %}{% endblock %}</title>
         <link rel="stylesheet" href="{% static "bootstrap/dist/css/bootstrap.css" %}" type="text/css" media="screen">
         <script type="text/javascript" src="{% static "jquery/dist/jquery.js" %}"></script>
+        <script type="text/javascript" src="{% static "jquery/dist/jquery-ui.js" %}"></script>
         <script type="text/javascript" src="{% static "bootstrap/dist/js/bootstrap.js" %}"></script>
         <link rel="stylesheet" href="{% static "schedule.css" %}" type="text/css" media="screen">
         {% block extra_head %}
```

Now everything works except choosing "This" (Choosing "All" or "Do nothing" works) when editing or deleting a recurring event:

```
AttributeError at /schedule/occurrence/edit/1/2017/1/2/8/0/0/
'NoneType' object has no attribute 'event' 
```

I haven't been troubleshooting this last error anything but it shouldn't be too hard finding root cause either for someone who is interested.

Christian

**Comment by llazzaro** [comment]
All js code in templates was deleted since it was duplicated.
I will add those function to .js file.
Also the jquery-ui dep is missing.

**Comment by HenryMehta** [comment]
I know this is closed but I am experiencing this issue.  I've not used django-bower before and so I am probably doing something wrong but I seem to have it installed and I am unable to delete or edit events.

I am using bootstrap and jquery elsewhere within the app and I wasn't using django-bower for these.  Is it conflicting and how do I rectify the situation.  You help is greatly appreciated

**Comment by llazzaro** [comment]
please can you send the output of "pip freeze" ?

**Comment by HenryMehta** [comment]
This is the output from pip freeze

appdirs==1.4.3
click==6.7
decorator==4.0.11
Django==1.11
django-annoying==0.10.3
django-bower==5.2.0
django-localflavor==1.4.1
django-registration==2.2
django-scheduler==0.8.3
geocoder==1.22.3
icalendar==3.11.3
olefile==0.44
packaging==16.8
Pillow==4.1.0
pkg-resources==0.0.0
pyparsing==2.2.0
python-dateutil==2.6.0
pytz==2017.2
ratelim==0.1.6
requests==2.13.0
simplejson==3.10.0
six==1.10.0
stripe==1.53.0

**Comment by 101pagejimmy** [comment]
Has this ever been resolved? I am using Python 3.6 and Django 1.11 and I am still receiving the same issues when attempting to delete single occurrences (i.e - 'NoneType' object has no attribute 'event'). This is after switching from Django 1.8 and Python 2.7 to attempt to fix the problem but to no avail. Even after downloading the django-scheduler example from git I had the same issue when attempting to delete single occurrences, being this, I don't believe the issue is on my side as it seems no matter what I do, deleting single occurrences leads to the same problem. Has anybody figured out a solution?
